File: program/check-palindrome-string/check_palindrome_string.go
package main

import "fmt"

func isPalindrome(str string) bool {
	var reverse string
	for _, v := range str {
		reverse = string(v) + reverse
	}
	return str == reverse
}

func main() {
	var input string

	fmt.Print("Input String: ")
	fmt.Scan(&input)

	if (isPalindrome(input)) {
		fmt.Println("Palindrome String")
	} else {
		fmt.Println("Not Palindrome String")
	}
}

File: program/program/check-palindrome-string/CheckPalindromeString.kt
fun main() {
    print("Input : ")
    println()
    var input = readLine()

    println("Is $input palindrome : " + CheckPalindromeString(input!!))

}

private fun CheckPalindromeString(inputStr: String): Boolean {
    val sb = StringBuilder(inputStr)

    //reverse the string
    val reverseStr = sb.reverse().toString()

    //compare reversed string with input string
    return inputStr.equals(reverseStr, ignoreCase = true)
}
